File: LeetCode/Swift/anagramMappings.swift
/*
    760. Find Anagram Mappings
    https://leetcode.com/problems/find-anagram-mappings/
 */

class Solution {
    func anagramMappings(_ A: [Int], _ B: [Int]) -> [Int] {
        var map = [Int : Int]()
        for n in 0..<B.count {
            map[B[n]] = n;
        }
        var res: [Int] = [];
        for n in 0..<A.count {
            res.append(map[A[n]]!);
        }
        return res;
    }
}

/*
    Runtime: 1 ms, faster than 97.78% of Java online submissions for Find Anagram Mappings.
    Memory Usage: 36 MB, less than 92.31% of Java online submissions for Find Anagram Mappings.
 */

File: LeetCode/Java/anagramMappings.java
/*
    760. Find Anagram Mappings
    https://leetcode.com/problems/find-anagram-mappings/
 */

class Solution {
    public int[] anagramMappings(int[] A, int[] B) {
        HashMap<Integer, Integer> map = new HashMap<>();
        for (int i = 0; i < A.length; i++)
            map.put(B[i], i);
        int[] res = new int[A.length];
        for (int i = 0; i < B.length; i++)
            res[i] = map.get(A[i]);
        return res;
    }
}

/*
    Runtime: 1 ms, faster than 97.78% of Java online submissions for Find Anagram Mappings.
    Memory Usage: 36 MB, less than 92.31% of Java online submissions for Find Anagram Mappings.
 */
